src/shared/orgUtils.ts

```diff
@@ -43,6 +51,39 @@ export class OrgUtils {
     return undefined;
   }
 
+  /**
+   * Queries the org and returns a list of the lightning experience apps in the org that are visible to and accessible by the user.
+   *
+   * @param connection the connection to the org
+   * @returns a list of the lightning experience apps in the org that are visible to and accessible by the user.
+   */
+  public static async getLightningExperienceAppList(connection: Connection): Promise<AppDefinition[]> {
+    const results: AppDefinition[] = [];
+
+    const appMenuItemsQuery =
+      'SELECT Label,Description,Name FROM AppMenuItem WHERE IsAccessible=true AND IsVisible=TRUE';
+    const appMenuItems = await connection.query<{ Label: string; Description: string; Name: string }>(
+      appMenuItemsQuery
+    );
+
+    const appDefinitionsQuery = "SELECT DeveloperName,DurableId FROM AppDefinition WHERE UiType='Lightning'";
+    const appDefinitions = await connection.query<{ DeveloperName: string; DurableId: string }>(appDefinitionsQuery);
+
+    appMenuItems.records.forEach((item) => {
+      const match = appDefinitions.records.find((definition) => definition.DeveloperName === item.Name);
+      if (match) {
+        results.push({
+          DeveloperName: match.DeveloperName,
+          Label: item.Label,
+          Description: item.Description,
+          DurableId: match.DurableId,
+        });
+      }
+    });
```

Comment:
Ideally we would do all of this in a single SQL statement using `JOIN` but as I understand it SF SOQL statements don't support `JOIN` so instead I run 2 query and filter in code (since it won't be the case that an org would have thousands and thousands of apps... but rather a handful of apps)